src/main/webapp/app/home/home.component.ts

```diff
@@ -0,0 +1,79 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, Inject, OnInit, } from '@angular/core';
+import { Task } from 'app/tasks/task';
+import { TaskService } from 'app/tasks/task.service';
+
+@Component({
+  selector: 'tiny-home',
+  templateUrl: './home.component.html',
+  styleUrls: ['./home.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class HomeComponent implements OnInit {
+  constructor(
+    @Inject('TaskService') private taskService: TaskService,
+    private ref: ChangeDetectorRef
+  ) {
+  }
+
+  private _filteredTasks: Task[] = [];
+  public get filteredTasks(): Task[] {
+    return this._filteredTasks;
+  }
+
+  public set filteredTasks(value: Task[]) {
+    this._filteredTasks = value;
+    this.ref.detectChanges();
+  }
+
+  public tasks: Task[] = [];
+
+  public errorMessage = '';
+
+  private _listFilter = '';
+  public get listFilter(): string {
+    return this._listFilter;
+  }
+
+  public set listFilter(value: string) {
+    this._listFilter = value;
```

Comment:
Totally agree. That would be a good point of improvement to reduce the amount of requests sent to the server. However, I didn't add such refinements to the code because it wasn't needed for the our case here. But if we were to implement this feature in a fully-working app, definitely that would be a great optimization.